src/NuGet.Services.AzureSearch/Catalog2AzureSearch/CatalogIndexActionBuilder.cs

```diff
@@ -312,30 +329,42 @@ private bool IsUpdateLatest(SearchIndexChangeType changeType)
             NuGetVersion version,
             HijackDocumentChanges changes)
         {
+            IndexAction<KeyedDocument> indexAction;
+
             if (changes.Delete)
             {
```

Comment:
Would a switch work better ?

Reply:
This is a series of bools and in another method we call a method in the `if` so I don't think it would help much.